source/helpers/selector-observer.tsx

```diff
@@ -46,7 +48,7 @@ export default function observe<
 	}
 
 	const selector = typeof selectors === 'string' ? selectors : selectors.join(',\n');
-	const seenMark = 'rgh-seen-' + getCallerID();
+	const seenMark = 'rgh-seen-' + getCallerID(ancestor);
```

Comment:
I don’t even know  how this worked until now. The observer's callback was only called once because the generated `rgh-seen` class name was identical and successive callbacks would fail.

The observer should likely be refactored to make sure all listeners are called (probably just needs a `await Promise.resolve()` before removing the class but alas)